blockprotocol/blockprotocol
# Only scroll users to anchor when certain conditions are met

## What this Does

The current behavior for scroll when moving between the homepage, and the `/spec` page of the blockprotocol website results in the user being scrolled to the first anchor. This results in the user missing key information including headers and a GitHub callout.

This PR attempts to fix this issue by only allowing a frontend scroll when certain conditions are met.

Changes are described in GH comments below.

## How to Test this

1. Go to the homepage and navigate to the spec page. This shouldn't cause a scroll down.
2. Clicking on sidebar items should scroll as normal.
3. Opening a # URL in any tab should scroll to the anchor normally. 
4. On clicking a top-level heading of a different page from the sidebar, scroll to the top, as it already does in the website.

> Note, in the existing code, an empty hash `#` and no hash are treated the same.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201668052739232
